hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataJavaResolver.java

```diff
@@ -240,7 +240,12 @@ private KvMetadata resolveObjectMetadata(
         );
     }
 
-    private static Class<?> loadClass(Map<String, String> options, boolean isKey) {
+    public static Class<?> loadClass(Map<String, String> options, Boolean isKey) {
+        if (isKey == null) {
+            String className = options.get(OPTION_TYPE_JAVA_CLASS);
+            return className != null ? loadClass(className) : null;
```

Comment:
is lack of classname valid here? It is hard to follow the flow, but won't that cause NPE?

Reply:
after `isKey` change, there is no longer potential NPE here. IMO worth adding the description in the comment, somewhere near `JavaEnricher. getSchemaId`

Reply:
9a0e8973